include/LightGBM/c_api.h

```diff
@@ -2,6 +2,7 @@
 #define LIGHTGBM_C_API_H_
 
 #include <LightGBM/meta.h>
```

Comment:
Is this file supposed to expose a c-api, which can be used by a C program?
If so, including header files with C++ code will not compile. I believe both - meta.h and log.h - have C++ code?
Other C++ stdlib headers (cstdint, exception, etc) included in this file would also not compile.

Reply:
We don't think it is used for c program.

Reply:
Forgive my ignorance. Why is it called "C" API then, when it cannot be used from a "C" program?

Reply:
that is for python package abd other language to call. Anyway, I just reduce the dependence of head files.